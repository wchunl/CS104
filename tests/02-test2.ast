ROOT "" (0.0.0)
|   TOK_STRUCT "struct" (4.1.1)
|   |   TOK_IDENT "node" (4.1.8) variable
|   |   TOK_INT "int" (4.2.5) int
|   |   |   TOK_TYPE_ID "foo" (4.2.9) typeid
|   |   TOK_PTR "ptr" (4.3.5)
|   |   |   TOK_TYPE_ID "node" (4.3.17) typeid
|   |   |   TOK_TYPE_ID "link" (4.3.23) typeid
|   TOK_FUNCTION "" (4.5.1) function
|   |   TOK_TYPE_ID "" (4.5.1) typeid
|   |   |   TOK_PTR "ptr" (4.5.1)
|   |   |   |   TOK_TYPE_ID "node" (4.5.13) typeid
|   |   |   TOK_IDENT "func" (4.5.19) variable
|   |   TOK_PARAM "(" (4.5.24) param
|   |   |   TOK_TYPE_ID "" (4.5.25) typeid
|   |   |   |   TOK_PTR "ptr" (4.5.25)
|   |   |   |   |   TOK_TYPE_ID "node" (4.5.37) typeid
|   |   |   |   TOK_IDENT "head" (4.5.43) variable (4.5.25)
|   |   |   TOK_TYPE_ID "" (4.5.49) typeid
|   |   |   |   TOK_INT "int" (4.5.49) int
|   |   |   |   TOK_IDENT "length" (4.5.53) int variable (4.5.49)
|   |   TOK_BLOCK "{" (4.5.61)
|   |   |   TOK_VARDECL "=" (4.6.11) variable lval
|   |   |   |   TOK_INT "int" (4.6.5) int
|   |   |   |   TOK_IDENT "a" (4.6.9) int variable (4.6.11)
|   |   |   |   TOK_INTCON "0" (4.6.13) int const
|   |   |   TOK_VARDECL "=" (4.7.14) variable lval
|   |   |   |   TOK_STRING "string" (4.7.5) string
|   |   |   |   TOK_IDENT "b" (4.7.12) string variable (4.7.14)
|   |   |   |   TOK_STRINGCON """" (4.7.16) string const
|   |   |   TOK_VARDECL "=" (4.8.25) variable lval
|   |   |   |   TOK_PTR "ptr" (4.8.5)
|   |   |   |   |   TOK_TYPE_ID "node" (4.8.17) typeid
|   |   |   |   TOK_IDENT "c" (4.8.23) struct variable (4.8.25)
|   |   |   |   TOK_ALLOC "alloc" (4.8.27)
|   |   |   |   |   TOK_TYPE_ID "node" (4.8.41) typeid
|   |   |   TOK_IF "if" (4.9.5)
|   |   |   |   '<' "<" (4.9.11) int
|   |   |   |   |   TOK_IDENT "a" (4.9.9) int variable (4.6.11)
|   |   |   |   |   TOK_INTCON "3" (4.9.13) int const
|   |   |   |   TOK_BLOCK "{" (4.9.16)
|   |   |   |   |   '=' "=" (4.10.10) int vreg
|   |   |   |   |   |   TOK_IDENT "a" (4.10.9) int variable (4.6.11)
|   |   |   |   |   |   TOK_IDENT "length" (4.10.11) int variable (4.5.49)
|   |   |   |   |   '=' "=" (4.11.10) int vreg
|   |   |   |   |   |   TOK_IDENT "c" (4.11.9) struct variable (4.8.25)
|   |   |   |   |   |   TOK_ARROW "->" (4.11.12) lval vaddr
|   |   |   |   |   |   |   TOK_IDENT "c" (4.11.11) struct variable (4.8.25)
|   |   |   |   |   |   |   TOK_IDENT "link" (4.11.14) variable
|   |   |   |   TOK_IFELSE "{" (4.12.11)
|   |   |   |   |   TOK_IF "if" (4.13.9)
|   |   |   |   |   |   TOK_EQ "==" (4.13.15) vreg
|   |   |   |   |   |   |   TOK_INTCON "0" (4.13.13) int const
|   |   |   |   |   |   |   TOK_INTCON "0" (4.13.18) int const
|   |   |   |   |   |   '=' "=" (4.13.23) int vreg
|   |   |   |   |   |   |   TOK_IDENT "a" (4.13.21) int variable (4.6.11)
|   |   |   |   |   |   |   TOK_INTCON "8" (4.13.25) int const
|   |   |   |   |   |   TOK_IFELSE "=" (4.14.16)
|   |   |   |   |   |   |   TOK_IDENT "b" (4.14.14) string variable (4.7.14)
|   |   |   |   |   |   |   TOK_STRINGCON ""foo"" (4.14.18) string const
