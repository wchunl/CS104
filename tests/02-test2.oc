struct node {
    int foo;
    ptr <struct node> link;
};
ptr <struct node> func (ptr <struct node> head, int length) {
    int a = 0;
    string b = "";
    ptr <struct node> c = alloc <struct node>();
    if (a < 3) {
        a=length;
        c=c->link;
    }else {
        if (0 == 0) a = 8;
        else b = "foo";
    }
}
