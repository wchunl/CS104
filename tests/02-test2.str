string_set[   0]:    9782517693237612205 0xb8d1f8->"func"
                    4706206694200543916 0xb8d4c8->","
string_set[   1]:    2297668033614959926 0xb8d978->"0"
string_set[   3]:    8248777770799913213 0xb8e2a8->"3"
string_set[   4]:   10838281452030117757 0xb8db28->"b"
                      21606623943142546 0xb8ca88->"{"
string_set[   6]:   13686721134187162920 0xb8cd28->"}"
string_set[   9]:   14526079891305879776 0xb8cf18->">"
string_set[  11]:    2699919699724185070 0xb8cf88->"link"
string_set[  12]:   15816506932648033945 0xb8d248->"("
                    3505610083221236292 0xb8d5d8->"length"
string_set[  13]:   14001456780366172028 0xb8d438->"head"
string_set[  14]:    9317282556992397945 0xb8ddd8->"if"
string_set[  16]:   13945322250941355849 0xb88318->"struct"
string_set[  17]:   17899137375351314596 0xb8ebb8->"8"
                   17176659130743002837 0xb8cd08->"ptr"
string_set[  19]:   10959529184379665549 0xb8ddb8->"c"
string_set[  22]:   14805121035671447500 0xb8e728->"->"
                    9369158146635886526 0xb8d038->""""
                    9631199822919835226 0xb8cc18->"foo"
string_set[  23]:   12868445110721718657 0xb8d458->")"
string_set[  24]:    4142143239838762426 0xb8eda8->""foo""
string_set[  25]:    4993892634952068459 0xb8d818->"a"
string_set[  31]:   14736352883882276958 0xb8eaa8->"=="
                    6142509188972423790 0xb88738->""
string_set[  32]:   11597697714117577063 0xb8d8c8->"="
string_set[  33]:    8290819294854170237 0xb8dfd8->"alloc"
string_set[  38]:   12597418120160561823 0xb8cb38->"int"
string_set[  39]:    1873634190020599143 0xb8e288->"else"
string_set[  42]:    5344662657899890615 0xb8cce8->";"
string_set[  44]:   13497578857207247412 0xb8da28->"string"
                   18345866965089363306 0xb88758->"node"
string_set[  45]:    1396094175042499165 0xb8cda8->"<"
load_factor = 0.702
bucket_count = 47
max_bucket_size = 3
