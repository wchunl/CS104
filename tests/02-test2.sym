
node (4.1.1) {0} struct node
   foo (4.2.5)  int field 0
   node (4.3.5) ptr <struct node> field 1
func (4.5.1) {0} ptr <struct node> function
   head (4.5.25) {1} ptr <struct node> variable param lval 0
   length (4.5.49) {1} int variable param lval 1
   a (4.6.11) {1} int variable local lval 0
   b (4.7.14) {1} string variable local lval 1
   c (4.8.25) {1} ptr <struct node> variable local lval 2
