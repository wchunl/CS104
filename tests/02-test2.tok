#  1 "02-test2.oc"
#  1 "<built-in>"
#  1 "<command-line>"
#  1 "02-test2.oc"
   4 1.00001 269 TOK_STRUCT    struct
   4 1.00008 282 TOK_IDENT     node
   4 1.00013 123 '{'           {
   4 2.00005 262 TOK_INT       int
   4 2.00009 282 TOK_IDENT     foo
   4 2.00012  59 ';'           ;
   4 3.00005 274 TOK_PTR       ptr
   4 3.00009  60 '<'           <
   4 3.00010 269 TOK_STRUCT    struct
   4 3.00017 282 TOK_IDENT     node
   4 3.00021  62 '>'           >
   4 3.00023 282 TOK_IDENT     link
   4 3.00027  59 ';'           ;
   4 4.00001 125 '}'           }
   4 4.00002  59 ';'           ;
   4 5.00001 274 TOK_PTR       ptr
   4 5.00005  60 '<'           <
   4 5.00006 269 TOK_STRUCT    struct
   4 5.00013 282 TOK_IDENT     node
   4 5.00017  62 '>'           >
   4 5.00019 282 TOK_IDENT     func
   4 5.00024  40 '('           (
   4 5.00025 274 TOK_PTR       ptr
   4 5.00029  60 '<'           <
   4 5.00030 269 TOK_STRUCT    struct
   4 5.00037 282 TOK_IDENT     node
   4 5.00041  62 '>'           >
   4 5.00043 282 TOK_IDENT     head
   4 5.00047  44 ','           ,
   4 5.00049 262 TOK_INT       int
   4 5.00053 282 TOK_IDENT     length
   4 5.00059  41 ')'           )
   4 5.00061 123 '{'           {
   4 6.00005 262 TOK_INT       int
   4 6.00009 282 TOK_IDENT     a
   4 6.00011  61 '='           =
   4 6.00013 283 TOK_INTCON    0
   4 6.00014  59 ';'           ;
   4 7.00005 263 TOK_STRING    string
   4 7.00012 282 TOK_IDENT     b
   4 7.00014  61 '='           =
   4 7.00016 285 TOK_STRINGCON ""
   4 7.00018  59 ';'           ;
   4 8.00005 274 TOK_PTR       ptr
   4 8.00009  60 '<'           <
   4 8.00010 269 TOK_STRUCT    struct
   4 8.00017 282 TOK_IDENT     node
   4 8.00021  62 '>'           >
   4 8.00023 282 TOK_IDENT     c
   4 8.00025  61 '='           =
   4 8.00027 273 TOK_ALLOC     alloc
   4 8.00033  60 '<'           <
   4 8.00034 269 TOK_STRUCT    struct
   4 8.00041 282 TOK_IDENT     node
   4 8.00045  62 '>'           >
   4 8.00046  40 '('           (
   4 8.00047  41 ')'           )
   4 8.00048  59 ';'           ;
   4 9.00005 265 TOK_IF        if
   4 9.00008  40 '('           (
   4 9.00009 282 TOK_IDENT     a
   4 9.00011  60 '<'           <
   4 9.00013 283 TOK_INTCON    3
   4 9.00014  41 ')'           )
   4 9.00016 123 '{'           {
   4 10.00009 282 TOK_IDENT     a
   4 10.00010  61 '='           =
   4 10.00011 282 TOK_IDENT     length
   4 10.00017  59 ';'           ;
   4 11.00009 282 TOK_IDENT     c
   4 11.00010  61 '='           =
   4 11.00011 282 TOK_IDENT     c
   4 11.00012 272 TOK_ARROW     ->
   4 11.00014 282 TOK_IDENT     link
   4 11.00018  59 ';'           ;
   4 12.00005 125 '}'           }
   4 12.00006 266 TOK_ELSE      else
   4 12.00011 123 '{'           {
   4 13.00009 265 TOK_IF        if
   4 13.00012  40 '('           (
   4 13.00013 283 TOK_INTCON    0
   4 13.00015 275 TOK_EQ        ==
   4 13.00018 283 TOK_INTCON    0
   4 13.00019  41 ')'           )
   4 13.00021 282 TOK_IDENT     a
   4 13.00023  61 '='           =
   4 13.00025 283 TOK_INTCON    8
   4 13.00026  59 ';'           ;
   4 14.00009 266 TOK_ELSE      else
   4 14.00014 282 TOK_IDENT     b
   4 14.00016  61 '='           =
   4 14.00018 285 TOK_STRINGCON "foo"
   4 14.00023  59 ';'           ;
   4 15.00005 125 '}'           }
   4 16.00001 125 '}'           }
