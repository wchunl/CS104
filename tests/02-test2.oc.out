; 0.  1: # 1 "02-test2.oc"; 1.  0: 
; 1.  1: # 1 "<built-in>"
; 2.  1: # 1 "<command-line>"
; 3.  1: # 1 "02-test2.oc"
; 4.  1: struct node {
; 4.  2:     int foo;
; 4.  3:     ptr <struct node> link;
; 4.  4: };
; 4.  5: ptr <struct node> func (ptr <struct node> head, int length) {
; 4.  6:     int a = 0;
; 4.  7:     string b = "";
; 4.  8:     ptr <struct node> c = alloc <struct node>();
; 4.  9:     if (a < 3) {
; 4. 10:         a=length;
; 4. 11:         c=c->link;
; 4. 12:     }else {
; 4. 13:         if (0 == 0) a = 8;
; 4. 14:         else b = "foo";
; 4. 15:     }
; 4. 16: }
